Loop/Public/FeedbackLoop.swift

```diff
@@ -453,6 +453,17 @@ extension Loop {
             }
         }
 
+        public func autoconnect(followingChangesIn predicate: @escaping (State) -> Bool) -> Feedback {
+            return Feedback { state, consumer in
+                self.events(
+                    state
+                        .skipRepeats { predicate($0.0) == predicate($1.0) }
+                        .flatMap(.latest) { predicate($0.0) ? state.filter { predicate($0.0) } : .empty },
+                    consumer
+                )
+            }
+        }
+
```

Comment:
This is what I managed to come up with, and it appears to be working. However I am not convinced this is the best approach... 🙈  

More specifically, I had to add that `filter` on the inner `state` to avoid the "disconnect-triggering" state from being processed before the inner chain is replaced.

Any ideas to make this tidier and/or more idiomatic?